ports/stm32/sdram.c

```diff
@@ -254,52 +253,113 @@ static void sdram_init_seq(SDRAM_HandleTypeDef
     #endif
 }
 
-bool sdram_test(bool fast) {
+void sdram_enter_low_power() {
+    // Enter self-refresh mode.
+    // In self-refresh mode the SDRAM retains data with external clocking.
+    FMC_SDRAM_CommandTypeDef command;
+    command.CommandMode = FMC_SDRAM_CMD_SELFREFRESH_MODE;
+    command.CommandTarget = FMC_SDRAM_CMD_TARGET_BANK;
+    command.AutoRefreshNumber = 1;
+    command.ModeRegisterDefinition = 0;
+    HAL_SDRAM_SendCommand(&hsdram, &command, 0xFFFF);
```

Comment:
Is there an LL HAL version of this that doesn't need the hsdram struct?

Reply:
Yes, hsdram is not actually needed, this is the F7 function:
```  __IO uint32_t tmpr = 0;
  
  /* Check the parameters */
  assert_param(IS_FMC_SDRAM_DEVICE(Device));
  assert_param(IS_FMC_COMMAND_MODE(Command->CommandMode));
  assert_param(IS_FMC_COMMAND_TARGET(Command->CommandTarget));
  assert_param(IS_FMC_AUTOREFRESH_NUMBER(Command->AutoRefreshNumber));
  assert_param(IS_FMC_MODE_REGISTER(Command->ModeRegisterDefinition));  

  /* Set command register */
  tmpr = (uint32_t)((Command->CommandMode)                  |\
                    (Command->CommandTarget)                |\
                    (((Command->AutoRefreshNumber)-1) << 5) |\
                    ((Command->ModeRegisterDefinition) << 9)
                    );
    
  Device->SDCMR = tmpr;
  
```
The H7 is pretty much the same, so I can resolve this issue and the global handle issue.